src/main/java/io/reactivex/internal/operators/flowable/FlowableFlattenIterable.java

```diff
@@ -0,0 +1,434 @@
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.flowable;
+
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.*;
+
+import org.reactivestreams.*;
+
+import io.reactivex.functions.*;
+import io.reactivex.internal.functions.Objects;
+import io.reactivex.internal.fuseable.QueueSubscription;
+import io.reactivex.internal.queue.SpscArrayQueue;
+import io.reactivex.internal.subscriptions.*;
+import io.reactivex.internal.util.*;
+import io.reactivex.plugins.RxJavaPlugins;
+
+public final class FlowableFlattenIterable<T, R> extends FlowableSource<T, R> {
+
+    final Function<? super T, ? extends Iterable<? extends R>> mapper;
+
+    final int prefetch;
+
+    public FlowableFlattenIterable(Publisher<T> source,
+            Function<? super T, ? extends Iterable<? extends R>> mapper, int prefetch) {
+        super(source);
+        if (prefetch <= 0) {
+            throw new IllegalArgumentException("prefetch > 0 required but it was " + prefetch);
+        }
+        this.mapper = Objects.requireNonNull(mapper, "mapper");
+        this.prefetch = prefetch;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void subscribeActual(Subscriber<? super R> s) {
+        if (source instanceof Callable) {
+            T v;
+
+            try {
+                v = ((Callable<T>)source).call();
+            } catch (Throwable ex) {
+                Exceptions.throwIfFatal(ex);
+                EmptySubscription.error(ex, s);
+                return;
+            }
+
+            if (v == null) {
+                EmptySubscription.complete(s);
+                return;
+            }
+
+            Iterator<? extends R> it;
+
+            try {
+                Iterable<? extends R> iter = mapper.apply(v);
+
+                it = iter.iterator();
+            } catch (Throwable ex) {
+                Exceptions.throwIfFatal(ex);
+                EmptySubscription.error(ex, s);
+                return;
+            }
+
+            FlowableFromIterable.subscribe(s, it);
+
+            return;
+        }
+        source.subscribe(new FlattenIterableSubscriber<T, R>(s, mapper, prefetch));
+    }
+
+    static final class FlattenIterableSubscriber<T, R>
+    extends BasicIntQueueSubscription<R>
+    implements Subscriber<T> {
+
+        /** */
+        private static final long serialVersionUID = -3096000382929934955L;
+
+        final Subscriber<? super R> actual;
+
+        final Function<? super T, ? extends Iterable<? extends R>> mapper;
+
+        final int prefetch;
+
+        final int limit;
+
+        final AtomicLong requested;
+
+        Subscription s;
+
+        Queue<T> queue;
+
+        volatile boolean done;
+
+        volatile boolean cancelled;
+
+        final AtomicReference<Throwable> error;
+
+        Iterator<? extends R> current;
+
+        int consumed;
+
+        int fusionMode;
```

Comment:
No. Saves a write and one has to be carefur not redefine `QueueSubscription.NONE`.